lib/join/server_token.go

```diff
@@ -39,8 +40,14 @@ func (s *Server) handleTokenJoin(
 	// Set any diagnostic info from the ClientParams.
 	setDiagnosticClientParams(stream.Diagnostic(), &tokenInit.ClientParams)
 
-	// There are no additional checks for the token join method, just make the
-	// result message and return it.
+	// When using a scoped token, ensure the TokenInit message contains a
+	// matching secret value.
+	if tok, ok := token.(*joining.Token); ok {
```

Comment:
Is there any way to achieve this validation without the type assertion?

Reply:
Just added a `GetSecret()` method to the `provision.Token` interface :+1:

Only slightly weird thing about that is the `ProvisionTokenV2` will return an empty string when calling `GetSecret()` so that agents don't need to be updated to include the token value in the `TokenInit` message. Which would be redundant anyway since the token name is already provided in the `ClientInit` message